### DISCUSSION TITLE
shadow-cljs Websocket connection failed

### DISCUSSION THREAD

**Discussion by surreal6** [uxbox/uxbox#498] [Q&A] [answered]
First of all, thank you very much for your initiative.

The project has motivated me to try learn a bit of clojure by looking at your code, but I have no experience with backend stuff, nor docker.

So I have some problem starting the exporter, this is the error: 

`penpot@e830115be68b:~/penpot/exporter$ ./scripts/wait-and-start.sh
[app.core] {:msg "initializing", :line 17}
The shadow-cljs Websocket was disconnected.
shadow-cljs - remote-error Error: connect ECONNREFUSED 127.0.0.1:9630
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 9630
}
[app.http] {:msg "starting http server", :port 6061, :line 16}
Stale Output! Your loaded JS was not produced by the running shadow-cljs instance. Is the watch for this build running?
^Cpenpot@e830115be68b:~/penpot/exporter$ node target/app.js
[app.core] {:msg "initializing", :line 17}
shadow-cljs - #3 ready!
[app.http] {:msg "starting http server", :port 6061, :line 16}
`

when manually executing `node target/app.js` as stated in the docs,  i get this:

`^Cpenpot@e830115be68b:~/penpot/exporter$ node target/app.js
[app.core] {:msg "initializing", :line 17}
shadow-cljs - #4 ready!
[app.http] {:msg "starting http server", :port 6061, :line 16}`

the browser can't connect to http://localhost:9630/ (where i suppose should appear the app) and http://localhost:6061/ shows just a 'Not found' message.

I have no clue on how to proceed, so I would appreciate a little help with it.

thanks in advance.

**Comment by niwinz** [comment, marked as answer]
Hello @surreal6 

Welcome!

You are currently runngin the devenv, our internal development enviroment. And you are having problem with the exporter start (we know that it does not works, it is just bad setup and we need to address it but for now is just low priority, because as you can see, if you manually run the `target/app.js` it starts as expected.

But don't get confused, **exporter** is just an internal module that is responsible of the export functionality; the **main application** starts at 3449 port (http://localhost:3449), and it does not depend on exporter to start. 

In any case if you just want to try it, you can consider using the docker-compose prepared just for this use case and is explained here: https://github.com/penpot/penpot/blob/develop/docs/00-Getting-Started.md

I hope this help you.

**Reply by surreal6** [reply]
Ok, it's working!! great!

Thank you very much @niwinz
